lib/ui/recommendation.js

```diff
@@ -263,22 +288,31 @@ Recommendation.prototype = {
     // duplicates one of the results, using nsIURI as a quick, easy way to
     // normalize and compare URLs.
     //
+    // If there's no duplicate, call this.show() to render the recommendation.
+    // Otherwise, return and nothing will be shown.
+    //
     // Comparison rules:
     // a wikipedia recommendation is a duplicate if a result has the same URL;
     // a top-level domain recommendation is a duplicate if any result is a page
     // from that domain (we only want to show new domains).
+    if (!this.searchResults || !this.rawData) {
+      return;
+    }
 
-    if (!this.searchResults || !this.recData) {
+    let type = this.getResultType(this.rawData);
+    this.data = this.transforms[type].transform(this.rawData);
```

Comment:
what if we wait to assign this.data until the dupe check inside show(), at line 172? we could pass the data to show() as a parameter. should fix the stale recommendation bug.